js/src/modals/AddContract/addContract.js

```diff
@@ -244,11 +247,26 @@ export default class AddContract extends Component {
         this.onClose();
       })
       .catch((error) => {
-        newError(error);
+        this.props.newError(error);
       });
   }
 
   onClose = () => {
     this.props.onClose();
   }
 }
+
+function mapStateToProps (state) {
+  return {};
+}
+
+function mapDispatchToProps (dispatch) {
+  return bindActionCreators({
+    newError
+  }, dispatch);
+}
+
+export default connect(
+  mapStateToProps,
```

Comment:
I generally prefer being explicit and in a once-off situation being explicit about an empty state is not bad. However who knows what happens with Redux making that call each cycle as things changes, it just add unnecessary overheads. 

Will update.